### PR TITLE
[#98] Refactor class components to hooks

### PR DESCRIPTION
**Pull request by kammanz**
Refactored the two class components, header and list, into function components.

What's completed:

- In header component, useEffect makes the getTeams call
- In list component, useEffect and useRef are used to scroll back to top on team select

### CODE REVIEW

**Comment by kammanz** [comment]
Ready for review.

File: src/components/header/index.js [hunk 1]
```diff
@@ -10,34 +10,27 @@ import NavMenu from './navMenu';
...
-    return (
-      <div className={styles.container}>
-        <DropdownMenu />
-        <NavMenu selectedTeam={selectedTeam} playerId={person_id} />
-        <UserLogin />
-      </div>
-    );
-  }
-}
+  }, []);
```
Comment (Tumne) [inline src/components/header/index.js]: As per our slack convo, your code is correct, the reason you see the warning is a side effect of `useEffect`. Essentially it's a scoping issue with the `getTeams`, `pathname`, and `history` params. The easiest thing would be to es lint disable the issue, but I don't think it'll be necessary.

Instead, if you move that function parameter into it's own const, it eliminates it cleanly:

```
  const getTeamsCallback = () => {
    getTeams(pathname, history);
  };

  useEffect(getTeamsCallback, []);
```

I know, it's dumb, but that's the way she goes. It's either solution 1 or solution 4 in this stackoverflow comment:

https://stackoverflow.com/a/60327893

File: src/components/players/list/index.js [hunk 2]
```diff
@@ -11,33 +11,27 @@ import { formatPlayerPhotoUrl } from 'utils/stringUtils';
...
-    const { selectedTeam: selectedTeamPrev } = prevProps;
-
-    if (selectedTeamPrev !== selectedTeam) {
-      this.ref.current.scrollTo(0, 0);
+  useEffect(() => {
+    if (selectedTeamRef.current !== selectedTeam) {
+      domRef.current.scrollTo(0, 0);
     }
-  }
+  });
```
Comment (Tumne) [inline src/components/players/list/index.js]: The concept behind `useEffect` is that there is no need to do a compare within the callback function: `if (selectedTeamRef.current !== selectedTeam)`

The proper usage would be to send in `selectedTeam` as a dependency. Then it acts like a `componentDidUpdate` and will compare `if (selectedTeamPrev !== selectedTeam)`

In that case we can simplify our useEffect to this:

```
  const domRef = useRef();

  useEffect(() => {
    domRef.current.scrollTo(0, 0);
  }, [selectedTeam]);
```

notice, that `selectedTeam` is the dependency injected and useEffect is listening to changes once `selectedTeam` changes.

**Review by Tumne — CHANGES_REQUESTED** [review]

File: src/components/players/list/index.js (hunk 2 above)
Comment (Tumne) [reply src/components/players/list/index.js]: Further refined, just use `teamId`:
```
  useEffect(() => {
    domRef.current.scrollTo(0, 0);
  }, [teamId]);
```

**Comment by kammanz** [comment]
Ready for review.

**Comment by kammanz** [comment]
Ready for review.

**Pull request merged**